pylint_django/checkers/migrations.py

```diff
@@ -0,0 +1,41 @@
+import astroid
+from pylint import interfaces
+from pylint.checkers import utils, BaseChecker
+from pylint_django.__pkginfo__ import BASE_ID
+
+def _is_migrations_module(node):
+    if not isinstance(node, astroid.Module):
+        return False
+
+    return 'migrations' in node.path[0] and not node.path[0].endswith('__init__.py')
+
+
+class MissingBackwardsMigrationChecker(BaseChecker):
+    __implements__ = (interfaces.IAstroidChecker,)
+
+    name = 'missing-backwards-migration-checker'
+
+    msgs = {'W%s05' % BASE_ID: ('Always include backwards migration callable',
+                                'missing-backwards-migration-callable',
+                                'Always include a backwards/reverse callable counterpart'
+                                ' so that the migration is not irreversable.')}
+
+    def visit_call(self, node):
+        try:
+            module = node.frame().parent
+        except:  # noqa: E722, pylint: disable=bare-except
+            return
+
+        if not _is_migrations_module(module):
+            return
+
+        if node.func.as_string().endswith('RunPython') and len(node.args) < 2:
+            if node.keywords:
+                for keyword in node.keywords:
+                    if keyword.arg == 'reverse_code':
+                        return
```

Comment:
I'm not sure what this block does, can you explain ?

Reply:
the reverse handler can also be passed into [`RunPython`](https://docs.djangoproject.com/en/3.0/ref/migration-operations/#runpython) via keyword argument `reverse_code` like in https://github.com/kiwitcms/Kiwi/blob/d7e456e4d56b15f262a7bade89bc90036590fc93/tcms/core/migrations/0001_squashed.py#L62 if its not via pos arg. So this is just for the checker to confirm if it is in the keyword arguments

Reply:
I see. This should be more robust then. Because I can also specify `code=forwards` as a keyword and then node.args becomes 0 length.

Here's the full signature of RunPython:
https://docs.djangoproject.com/en/3.0/ref/migration-operations/#django.db.migrations.operations.RunPython

Reply:
Wouldn't this check already catch that? I already had this in mind with the example I linked from the Kiwi repo.

Reply:
Nope. I can write my migration as such:
```
migrations.RunPython(code=forwards, reverse_code=backwards)
```

and node.args will be an empty list. However this is perfectly valid invocation.

Or I can call it like so:

```
migrations.RunPython(code=forwards, atomic=False)
```

and node.args will still be empty list but this invocation is missing the reverse_code argument.

Reply:
IMO a break-down of the if condition will simplify things, e.g.

1) if len(node.args) < 2 - definitely an error condition, report it and return from the method
2) if not then inspect kwargs for both code/reverse_code keywords taking into account that the first one can also be a positional argument.

Reply:
I've double-checked with your example, the linter will still catch it. The for loop on line 133 still won't find the reverse_code argument and the linter will bring up the issue.

Reply:
Add a test case for it then.

Reply:
Added your examples and the linter is working as expected.